### DISCUSSION TITLE
KEY and SECRET not fully explained

### DISCUSSION THREAD

**Discussion by adminradio** [directus/directus#17718] [Archive]
Sorry, I know it's asked a few times before, but I don't understand them fully in depth. My understanding so far is, that both are (unique) instance variables. That means, when we dump data from an instance A and restore the same data to an instance B, then there is no conflict when we did setup different values for SECRET and KEY on A and B. They are just ephemeral runtime environment variables and it doesn't hurt if they change - am I right?

**Comment by adminradio** [comment]
**Answering my own question for reference:** I just dumped a complete database from **instance A** and restored it into instance **B**. Running **B** by using different KEY and SECRET values as **A** does not break an values (like passwords or salted hashes). AFAIK.

**Comment by merlinaudio** [comment]
From [a discussion on Discord](https://discord.com/channels/725371605378924594/1105491909906010222):

- `KEY` is only used as an identifier in multi-container setups
- `SECRET` is used by the Directus API to sign JWTs for the Directus frontend
